File: Javascript/2051-2100/2079. Watering Plants.js
// https://leetcode.com/problems/watering-plants/

var wateringPlants = function(plants, capacity) {
    let steps = 0, water = capacity;

    for(let index = 0; index < plants.length; index++){
        if(plants[index] > water){
            steps += (index)*2;
            water = capacity; // Refill water
        }
        water -= plants[index];
        steps++;
    }

    return steps;
};

File: Python/2051-2100/2079. Watering Plants.py
# https://leetcode.com/problems/watering-plants/

class Solution:
    def wateringPlants(self, plants: List[int], capacity: int) -> int:
        steps, water = 0, capacity

        for index in range(0,len(plants)):
            if(plants[index] > water):
                steps += (index)*2
                water = capacity; # Refill water
            
            water -= plants[index]
            steps += 1

        return steps
